docs/docs/recognizers/strategy.md

```diff
@@ -0,0 +1,50 @@
+# Strategy
+## Introduction
+
+### Context of use
+The strategy pattern comes in handy when multiple methods do the same thing in different ways, this pattern is in this case used to reduce code duplication. The gang of four describes the Strategy Pattern in the following manner: “Define a family of algorithms, encapsulate each one, and make them interchangeable. Strategy lets the algorithm vary independently from clients that use it”. It enables the user to use different business rules or algorithms depending on the context in which they occur.
+
+### Example of use
+An example case that shows the need of the pattern is when someone uses a route calculator which calculates the shortest route from A to B. Someone then might want to use the strategy pattern for each of the different routing options (Bike, Car, etc). Then the code can just call a function CreateRoute independent of which routing algorithm is selected.
+
+## Explanation of Architecture
+![UML Strategy](https://refactoring.guru/images/patterns/diagrams/strategy/structure.png)
+
+The **Context** is the class from where the code gets executed. It maintains a reference to one of the concrete strategies and communicates with the object only via the Strategy interface.
```

Comment:
I would say: "It has a field of type Strategy in which a reference to one of the concrete strategies is stored."

Reply:
I changed it to: "The **Context** is the class from where the code gets executed. It stores a reference to one of the concrete strategies. It only communicates with the object via the methods in the Strategy." Because it doesn't have to be a field, it could also be a property